### PR TITLE
Prioritize release builds

### PR DESCRIPTION
**Pull request by kcgen**
Although release builds might be easy on some systems (say, a desktop linux distro with libraries for all dependencies), it's infact quite a bit more challenging for users on windows using MSYS2, Raspberry Pi users (via RetroPie), macOS users, and Linux users with OSes that might be very new or slightly out-of-date. 

For example:

``` text
 — 
Yesterday at 1:48 AM
I've ended up with > -Dbuildtype=release
-Dtry_static_libs=fluidsynth, iir,mt32emu, opusfile,png,
sdl2_net,slirp,speexdsp -Dfluidsynth:enable-floats=true
-Dfluidsynth:try-static-deps=true

I got error while loading shared libraries: libmt32emu.so
on running the binary (the external munt/mt32emu 2.6x
is properly downloaded and compiled) 
looks like tracy is being compiled in as well despite
not having it in the list.

 — 
Yesterday at 2:45 AM
still facing the mt32 related error

I've been trying all sort of combo without success.
I'm currently using this: -Dbuildtype=release
-Dtry_static_libs=iir,mt32emu -Dfluidsynth:enable-floats=true
-Dfluidsynth:try-static-deps=true.

also tried same options as from the main page w/o success either:
-Dbuildtype=release \
    -Ddefault_library=static \
    -Db_asneeded=true \
    -Dtry_static_libs=iir \ (same error if iir,mt32emu)
    -Dfluidsynth:enable-floats=true \
    -Dfluidsynth:try-static-deps=true 
tracy is always compiled and installed btw even when
passing -Dtracy=false 
```

The prior message feed actually starts around 1 pm, so 2 hours and 45 minutes were spent wrestling with our build flags. This definitely falls into severe paper-cuts category.

This PR prioritizes kitchen-sink static release builds as-default, which:
 - updates the Meson build and default settings
 - updates the build documentation (now much simpler) for both users and packagers
 - updates the CI jobs (simpler release builds), while debug builds need `Dbuildtype=debug` added
 - fixes a handful of static analysis issues and new compiler warnings that were now being discovered as a result of these slightly different builds (for example: our debugger builds now correctly default to release-mode, which generated most of these warnings).
 - updates tracy to only build and use it when requested
 - updates the speexDSP wrap, which [needed updating](https://github.com/mesonbuild/wrapdb/pull/547) to build properly under ARM.
 - finally, it cleans up the meson formatting using `muon fmt` (https://git.sr.ht/~lattis/muon)

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -97,19 +97,27 @@ up builds.
...
 
 Meson supports several build types, appropriate for various situations:
-`release` for creating optimized release binaries, `debug` (default) for
-for development or `plain` for packaging.
+`release` (default) for creating optimized release binaries, `debug` for
+for development.
+
+Packagers interested in using shared libraries can use
+`-Ddefault_library=shared` and `-Duse_system_libs=lib1,lib2,etc` to
+ensure sepected dependencies are provided by the system instead of
```
Comment (johnnovak) [inline BUILD.md:113]: typo: expected

File: BUILD.md [hunk 2]
```diff
@@ -97,19 +97,27 @@ up builds.
...
+via Meson wrap. See `meson_options.txt` for the full list.  Alternately,
+wraps can be fully disabled using `-Dwrap_mode=nofallback` if your
+operating system can provide all libraries for all of Staging's
+dependencies. For example:
```
Comment (johnnovak) [inline BUILD.md:117]: Nice doco, I was wondering myself how this all works when I build on macOS.

File: contrib/generate_tracy_header/header_top.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2022  The DOSBox Staging Team
```
Comment (johnnovak) [inline contrib/generate_tracy_header/header_top.txt:4]: 2022-2022, so it's easier to update later via scripts?

File: include/tracy.h [hunk 4]
```diff
@@ -0,0 +1,138 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2022  The DOSBox Staging Team
```
Comment (johnnovak) [inline include/tracy.h:4]: 2022-2022?

File: contrib/generate_tracy_header/header_top.txt (hunk 3 above)
Comment (kcgen) [reply contrib/generate_tracy_header/header_top.txt:4]: Ahh yes, thanks for catching that!

**Review by johnnovak — APPROVED** [review]
That's a heroic magnitude of changes and improvements, @kcgen. I have trouble meaningfully reviewing build related PRs, but skimmed through most of it and could only spot some very minor things.

File: BUILD.md (hunk 1 above)
Comment (kcgen) [reply BUILD.md]: thanks!

**Comment by kcgen** [comment]
Thanks, @johnnovak ! This bloated PR sure gave the GitHub CI boxes got a workout. 

I was hoping it would be more contained - however changing the build types (and getting more representative debugger and static analyzer builds) produced new warnings and issues, so it rippled from there.

All said, I'm much appier with our default builds; those massive command lines are hopefully a thing of the past.

**Pull request merged**